iothub_client/src/iothub_client_core_ll.c

```diff
@@ -615,6 +619,22 @@ static const char* IoTHubClientCore_LL_GetProductInfo(void* ctx)
     return result;
 }
 
+static const char* IoTHubClientCore_LL_GetDTModelId(void* ctx)
```

Comment:
>IoTHubClientCore_LL_GetDTModelId [](start = 19, length = 32)

The design here is generally right, but I wonder instead could you extend out IoTHubClientCore_LL_GetProductInfo() so that maybe it's like

int IoTHubClientCore_LL_GetProductInfo(char** product_ino, char** dt_model_id); // once we get model_id name agreed on.

This means there's just one callback to worry with and if hypothetically we added more fields to this later, it'd just be extended out this fcn instead of new ones.

In the MQTT layer, it would check the value(s) returned here and then do right thing.
?